uiowa/uiowa
# Place for some p2lb updates

Resolves https://github.com/uiowa/uiowa/issues/5254. 
Resolves #5318 





# How to test
(hawkeyemarchingband or brand are good v2 candidates to use, but feel free to use any v2 site)
`drush @hawkeyemarchingband.local en sitenow_p2lb`
`drush @hawkeyemarchingband.local config-split:activate p2lb`

If testing the conversion of ListBlock blocks (People, Articles), either the v2 split needs to then be disabled, or the following code needs to be updated locally to avoid the v2 check: https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/layout_builder_custom/layout_builder_custom.module#L1240 (eg manually set `$is_v2` to `FALSE` before the check).

Visit the P2Lb config page (/admin/config/sitenow/p2lb) or the converter page (/admin/sitenow-converter) and convert some existing pages or newly created test pages, and check their conversions.



## Updates to test:
- Articles
  - Articles block created is the views ListBlock version of an articles block, not the old Articles block
  - V2 title field is included in the v3 block, and display/hide values match
  - All default styles are set. You should be able to edit the block and immediately save without errors.
  - Pager use properly maps (hidden if hidden on v2, displayed if previously used), and does not error if the pager is used.
- Events
  - display/hide value of v3 headline matches display/hide value of v2
- People
  - display/hide value of v3 headline matches display/hide value of v2
  -   - Pager use properly maps (hidden if hidden on v2, displayed if previously used), and does not error if the pager is used.
- Featured Content:
  - display/hide value of v3 headline matches display/hide value of v2
- Image Gallery
  - display/hide value of v3 headline matches display/hide value of v2
- Webform:
  - v2 title is included as a v3 text area w/headline, placed above the webform (and display/hide value of the v3 headline matches the display/hide value of the v2 title)
- Accordion:
  - V2 title field is included in the v3 block, and display/hide values match
- Carousel/Slider:
  - V2 title field is included in the v3 block, and display/hide values match

If other issues are spotted, check that their is a corresponding note and/or issue in #3878 to account for it, and if not, it can be either addressed in this PR or added there for later followup.
